### PR TITLE
Hue binding: fix 1880

### PR DESCRIPTION
**Pull request by lolodomo**
Member bridgeHandler was not reset to null in dispose().

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java [hunk 1]
```diff
@@ -82,16 +82,23 @@
...
 
     public HueLightHandler(Thing hueLight) {
         super(hueLight);
+        lightId = null;
+        lastSentColorTemp = null;
+        lastSentBrightness = null;
+        isOsramPar16 = false;
+        propertiesInitializedSuccessfully = false;
+        bridgeHandler = null;
+        scheduledFuture = null;
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: What's the reasons for that initializations here? Shouldn't all that ones not be done implicit already?

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java [hunk 2]
```diff
@@ -160,7 +167,8 @@ public void dispose() {
         if (lightId != null) {
             HueBridgeHandler bridgeHandler = getHueBridgeHandler();
             if (bridgeHandler != null) {
-                getHueBridgeHandler().unregisterLightStatusListener(this);
+                this.bridgeHandler.unregisterLightStatusListener(this);
+                this.bridgeHandler = null;
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: To be consistent I would remove the "this." in front of bridgeHandler (it is also missed in the if clause already).

**Comment by maggu2810** [comment]
I agree, it should be set to null on dispose, because the initialize functions checks for != null.
Some minor comments added.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java (hunk 1 above)
Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: When you have just a declaration like that:
```
private HueBridgeHandler bridgeHandler;
```
Does it implicit init bridgeHandler to null ?

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java (hunk 2 above)
Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: getHueBridgeHandler() returns "this.bridgeHandler".
I have a doubt if using "bridgeHandler = null;" will set the member of the class to null ?

**Comment by lolodomo** [comment]
If bridgeHandler is not set to null when dispose() is called, the listener will be unregistered when dispose() is called but not registered again when initialize() is called. That was the issue.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java (hunk 1 above)
Comment (9037568) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: All reference types are null by default. See [JLS](http://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.3) sections 4.3 and 4.12.5.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java (hunk 2 above)
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: You are right, I have not realized the line above the if clause and missed that local member.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java (hunk 1 above)
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: As stated by @9037568 the chapter 4.12.5 contains a list of the default values:
http://docs.oracle.com/javase/specs/jls/se8/html/jls-4.html#jls-4.12.5

Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: Ok I will restore the previous code.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java (hunk 2 above)
Comment (lolodomo) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueLightHandler.java]: But for the call to unregister I could still use the local reference like before.
In fact I could just add one line and keep the old one unchanged.
Will change that.

**Comment by lolodomo** [comment]
Ok, PR updated.

**Comment by maggu2810** [comment]
Thanks @lolodomo.

**Pull request merged**